tools/wpt/runner.ts

```diff
@@ -0,0 +1,131 @@
+import { delay, readLines } from "../util.js";
+import { assert, ManifestTestOptions, release, runPy } from "./utils.ts";
+import { DOMParser } from "https://deno.land/x/deno_dom@v0.1.3-alpha2/deno-dom-wasm.ts";
+
+export async function runWithTestUtil<T>(
+  verbose: boolean,
+  f: () => Promise<T>,
+): Promise<T> {
+  const proc = runPy(["wpt", "serve"], {
+    stdout: verbose ? "inherit" : "piped",
+    stderr: verbose ? "inherit" : "piped",
+  });
+
+  const start = performance.now();
+  while (true) {
+    await delay(1000);
+    try {
+      const req = await fetch("http://localhost:8000/");
+      await req.body?.cancel();
+      if (req.status == 200) {
+        break;
+      }
+    } catch (err) {
+      // do nothing if this fails
+    }
+    const passedTime = performance.now() - start;
+    if (passedTime > 15000) {
+      throw new Error("Timed out while trying to start wpt test util.");
+    }
+  }
+
+  if (verbose) console.log(`Started wpt test util.`);
+
+  try {
+    return await f();
+  } finally {
+    if (verbose) console.log("Killing wpt test util.");
+    proc.kill(2);
+    await proc.status();
+    proc.close();
+  }
+}
+
+export interface TestResult {
+  cases: TestCaseResult[];
+  status: number;
+  stderr: string;
+}
+
+export interface TestCaseResult {
+  name: string;
+  passed: boolean;
+  status: number;
+  message: string | null;
+  stack: string | null;
+}
+
+export async function runSingleTest(
+  url: URL,
+  options: ManifestTestOptions,
+  reporter: (result: TestCaseResult) => void,
+): Promise<TestResult> {
+  const bundle = await generateBundle(url);
+  const tempFile = await Deno.makeTempFile({
+    prefix: "wpt-bundle-",
+    suffix: ".js",
+  });
+  await Deno.writeTextFile(tempFile, bundle);
+
+  const proc = Deno.run({
+    cmd: [
+      `./target/${release ? "release" : "debug"}/deno`,
```

Comment:
Resolve relative to current file? Otherwise it will fail if not run from root of the repo